src/sssom_schema/schema/sssom_schema.yaml

```diff
@@ -118,15 +119,17 @@ slots:
   subject_id:
     description: The ID of the subject of the mapping.
     range: EntityReference
-    required: true
```

Comment:
unfortunate side effect, need some help to ensuring that either this or `subject_literal` is required.. (@gouttegd generally feel free to just commit to this branch, we will recreate it nicely once the design is finished)

Reply:
I don’t think this is feasible in pure LinkML.¹ So it would have to be done on top of LinkML, by adding a paragraph to the spec to explicitly state that `subject_id` MUST have a value, unless the `subject_type` is `sssom literal`, in which case it is `subject_label` (or `subject_literal` if I did not push you enough to accept reusing `subject_label`) that MUST have a value.

Implementations that use the Python LinkML runtime may need to be updated to cope with that change, if they were previously relying on the runtime to check that `subject_id` was always set (I don’t know if that’s the case of SSSOM-Py). But for non-Python implementations, this will not change much, since the absence of a LinkML runtime for any other language than Python means that those implementations already had to “manually” enforce the `required` constraint anyway.

---
¹ Feel free to ask people who know LinkML better than I do (so, approximately everyone) just in case. But as far as I can tell, we cannot express variable constraints in LinkML, such as “this slot is required IFF that other slot has this particular value”.